src/vs/workbench/services/search/node/ripgrepTextSearchEngine.ts

```diff
@@ -495,9 +495,14 @@ function getRgArgs(query: TextSearchQuery, options: TextSearchOptions): string[]
 /**
  * `"foo/*bar/something"` -> `["foo", "foo/*bar", "foo/*bar/something", "foo/*bar/something/**"]`
  */
-function spreadGlobComponents(globArg: string): string[] {
-	const components = splitGlobAware(globArg, '/');
-	return components.map((_, i) => components.slice(0, i + 1).join('/'));
+function spreadGlobComponents(globComponent: string): string[] {
```

Comment:
As the glob processing gets more complicated, it would probably be helpful to break out the code in `getRgArgs` with the `!*` etc that handles glob patterns and write a unit test for it.

Reply:
for now, just added unit tests for that part of `getRgArgs`